ICAFIX/ApplyHandReClassifications.sh

```diff
@@ -4,140 +4,193 @@
 
-PARAMETERs are: [ ] = optional; < > = user supplied value
-  [--help] : show usage information and exit
-   --path=<path to study folder> OR --study-folder=<path to study folder>
-   --subject=<subject ID>
-   --fmri-name=<fMRI name>
-   --high-pass=<high-pass filter used in ICA+FIX>
+opts_SetScriptDescription "Apply Hand Reclassifications of Noise and Signal components from FIX using the ReclassifyAsNoise.txt and ReclassifyAsSignal.txt input files. Generates HandNoise.txt and HandSignal.txt as output. Script does NOT reapply the FIX cleanup. For that, use the ReApplyFix scripts."
 
-EOF
-}
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "path to study folder" "--path"
 
-# ------------------------------------------------------------------------------
-#  Get the command line options for this script.
-# ------------------------------------------------------------------------------
-get_options()
-{
-	local arguments=($@)
-
-	# initialize global output variables
-	unset p_StudyFolder
-	unset p_Subject
-	unset p_fMRIName
-	unset p_HighPass
-	g_matlab_run_mode=0
-
-	# parse arguments
-	local num_args=${#arguments[@]}
-	local argument
-	local index=0
-
-	while [ ${index} -lt ${num_args} ]; do
-		argument=${arguments[index]}
-
-		case ${argument} in
-			--help)
-				show_usage
-				exit 0
-				;;
-			--path=*)
-				p_StudyFolder=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--study-folder=*)
-				p_StudyFolder=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--subject=*)
-				p_Subject=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--fmri-name=*)
-				p_fMRIName=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--high-pass=*)
-				p_HighPass=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--matlab-run-mode=*)
-				g_matlab_run_mode=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			*)
-				show_usage
-				log_Err_Abort "unrecognized option: ${argument}"
-				;;
-		esac
-	done
-
-	local error_count=0
-
-	# check required parameters
-	if [ -z "${p_StudyFolder}" ]; then
-		log_Err "Study Folder (--path= or --study-folder=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_StudyFolder: ${p_StudyFolder}"
-	fi
+opts_AddMandatory '--subject' 'Subject' 'id' "subject ID"
 
-	if [ -z "${p_Subject}" ]; then
-		log_Err "Subject ID (--subject=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_Subject: ${p_Subject}"
-	fi
+opts_AddMandatory '--fmri-name' 'fMRIName' 'string' "fMRI name"
 
-	if [ -z "${p_fMRIName}" ]; then
-		log_Err "fMRI Name (--fmri-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_fMRIName: ${p_fMRIName}"
-	fi
+opts_AddMandatory '--high-pass' 'HighPass' 'amount' "high-pass filter used in ICA+FIX"
 
-	if [ -z "${p_HighPass}" ]; then
-		log_Err "High Pass: (--high-pass=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_HighPass: ${p_HighPass}"
-	fi
+##Optional Args 
+## MATLAB RUN MODE IS NOT ACTUALLY USED
```

Comment:
The usage explanation of this unused matlab mode argument should probably be deleted and replaced with "deprecated, matlab code is not used".

Reply:
Done

Reply:
No, you changed the comment, not the help text explanation.

Reply:
Actually, did you test this version of it?  With `set -eu`, the removed `G_DEFAULT_MATLAB_RUN_MODE` variable should have caused it to abort when it got to this description.